tools/pin_github_action.py

```diff
@@ -1,45 +1,30 @@
 from __future__ import annotations
 
 import argparse
-import json
-import os
 import re
-import urllib.request
+import subprocess
 from functools import lru_cache
 from typing import Sequence
-from urllib.error import HTTPError
 
 
 @lru_cache(maxsize=None)
-def get_sha(repo: str, ref: str, github_token: str) -> str:
+def get_sha(repo: str, ref: str) -> str:
     if len(ref) == 40:
         try:
             int(ref, 16)
-            return ref
         except ValueError:
             pass
+        else:
+            return ref
 
-    try:
-        resp = urllib.request.urlopen(
-            urllib.request.Request(
-                f"https://api.github.com/repos/{repo}/commits/{ref}",
-                method="GET",
-                headers={
-                    "Accept": "application/vnd.github+json",
-                    "Authorization": f"token {github_token}",
-                    # A user agent is required. Lol.
-                    "User-Agent": "python-requests/2.26.0",
-                },
-            )
-        )
-    except HTTPError as e:
-        print(f"Status {e.code} while resolving {ref} for {repo}.")
-        if e.code == 403:
-            print("You most likely didn't authorize your token for SAML to the getsentry org.")
-        return ref
-
-    data: dict[str, str] = json.load(resp)
-    return data["sha"]
+    cmd = ("git", "ls-remote", "--exit-code", f"https://github.com/{repo}", ref)
```

Comment:
Commands' help: `git-ls-remote - List references in a remote repository`

```
 --exit-code
           Exit with status "2" when no matching refs are found in the remote repository. Usually the command exits with status "0" to indicate it successfully talked with the remote
           repository, whether it found any matching refs.
```

Sample output (3 lines):
```text
4ee1af708e1e130e29bc46388c0a9192dcc02443	refs/reviewable/pr3836/r1
8850500135277823d37fbadabb04b202d738db74	refs/tags/v1.8.8
91c61b2c5cc09747a95fd1505d4df4f1f4fcf053	refs/tags/v1.8.8^{}
```